src/engine/IOrderedObject.hpp

```diff
@@ -0,0 +1,12 @@
+#pragma once
+#include <string>
+
+namespace admirals {
+class IOrderedObject {
+public:
+    IOrderedObject(const std::string &name, float order)
+        : m_name(name), m_order(order) {}
+    const std::string m_name;
+    const float m_order;
```

Comment:
The order needs a getter, and probably a setter(??), although that would clash with the OrderedCollection..